### DISCUSSION TITLE
Problem: Bulidung a Freetz-ng image does not work. I get ERROR: firmware image 7590-xx_labor.detected.image not found

### DISCUSSION THREAD

**Discussion by Herbie-2005** [Freetz-NG/freetz-ng#993] [Allgemein] [answered]
Hi!

Trying to build a Freetz-ng image, the buildung process always leads to the same error message:

ERROR: firmware image 7590-xx_labor.detected.image not found

I checked Freetz-ng from scratch, renewed prerequisites on Ubuntu 20.04.5 LTS, and tried with final image "7590-07.59" and with labor image "FRITZ.Box_7590-07.90-112649-LabBETA.image". The result is always the same: the above given error.

I removed the .freetz-dl folder completely, so it was created again. Starting the building process with "make", the corresponding firmware image gets downloaded into .freetz-dl, but afterwards I get the error "not found" again.

A few days ago, there was no problem at all.

Does anyone have an idea of what could be wrong?

**Comment by fda77** [comment]
maybe the quoting changes, please attach .config

**Comment by Herbie-2005** [comment]
@
[.config.zip](https://github.com/user-attachments/files/15685890/default.config.zip)

**Comment by fda77** [comment]
i have no idea why it did not work for you, this works:

```
git clone https://github.com/freetz-ng/freetz-ng ~/muh
mkdir -p ~/muh/dl
wget https://github.com/user-attachments/files/15685890/default.config.zip -O ~/muh/x.zip
unzip -p ~/muh/x.zip > ~/muh/.config
make firmware-nocompile     # dont flash a nocompile image!

STEP 3: PACK/SIGN
  checking for left over version-control-system files
  integrate freetz info file into image
packing var.tar
  checking signature key files
    adding public signature key file
creating filesystem image (SquashFS4_le-xz)
  SquashFS block size: 64 kB (65536 bytes)
copying kernel image
  kernel image size: 6.9 MB, max 8.0 MB, free 1.1 MB (1198592 bytes)
copying filesystem image
  filesystem image size: 32.5 MB, max 44.0 MB, free 11.5 MB (12058624 bytes)
adding checksum to kernel.image
adding checksum to filesystem.image
packing images/7590_07.90.all-rev112649_labor_freetz-ng-23786M-836352d45_20240606-222714.image
  packed image file size: 39.5 MB (41400320 bytes)
signing packed .image file
  signed image file size: 39.5 MB (41400320 bytes)
source firmware: 7590_de-es-fr-it-nl-pl 154.07.90-BETA rev112649 {ALL} [Smart24 P1] (25.04.2024 17:23:41)
  source image file size: 39.5 MB (41431040 bytes)
done.
```

btw, quoting changes : 784f3672b2cb2727b9b67e803a7f8dd3761ed62e

**Reply by Herbie-2005** [reply]
Thank you. What you did, does not work or me. I get the same error message as before.

**Comment by fda77** [comment]
i gave no idea what else could be wrong

**Reply by Herbie-2005** [reply]
Well, I can avoid the error, if I delete the quotation marks in Makefile, line 429.

**Comment by fda77** [comment]
i tested with xbuntu24 and fedora40, no idea, works!
can you find the line causing the error?

**Reply by fda77** [reply]
contains your image file name some silly characters? like " (1)" introduced by avm this week?
[config/mod/dl-firmware.in](https://github.com/Freetz-NG/freetz-ng/blob/master/config/mod/dl-firmware.in#L642)

**Reply by Herbie-2005** [reply]
As I wrote: Makefile, line 429. I could build a new Fretz-ng image now while quotation marks removed. -- No silly characters.

You just added quotation marks in 700-image.mk and (maybe) in Makefile, line 429, too. Is it possible that it leads to double quotation marks?

**Reply by fda77** [reply]
quotation marks are to put multiple word in 1 variable
```
./somecommand aaa bbb ccc      -> 3 args
./somecommand "aaa bbb ccc"      -> 1 arg
```

**Comment by fda77** [comment]
Sry :)

I removed them to test, and select 7530ax+labor+NOjuis:
```
bash: -c: line 14: syntax error near unexpected token `('
bash: -c: line 14: `    dl/fw/FRITZ.Box_7530_AX-07.90-113462-LabBETA (1).image'
make[1]: *** [Makefile:416: firmware-nocompile] Error 2
make: *** [Makefile:46: envira] Beendet

```

they are needed

**Reply by fda77** [reply]
Maybe replace `"` by  `'`  helps on your system?

**Reply by Herbie-2005** [reply]
No, it does not help.

What I found out: When I add a line in Makefile, just before line 415
    @echo "$(DL_IMAGE)"
I get the following output:
   " dl/fw/FRITZ.Box ..."
There is a space character at the beginning (directly before dl). I do not know, why. But could this be a problem?

**Comment by fda77** [comment]
im not very experienced with makefiles!
you could put somewhere:
```
xy:
<tab>@echo "whatever $(DL_IMAGE)"

and then run "make xy"
```

to remove surrounding whitespaces there is the `$(strip  `
but it could be there okay because of the `DL_IMAGE+=$$(IMAGE$(1))` which is for aliens with 2 or 3 images.
so quoting is wrong also for aliens

**Reply by Herbie-2005** [reply]
Well, I think it would best to find out why DL_IMAGE (under certain circumstances) starts with a space character and to remove it. Afterwards the path would be correct, especially with quotation marks.

**Comment by fda77** [comment]
because there could be multiple images in case of aliens, separated by comma! and new strings are added with a prefix-space `+=`

min 4 possible cases:
```
 -1 default: DL_IMAGE=firmware.image
 -2 alien: DL_IMAGE=firmware.image firmware_tk.image
 -3 labor: DL_IMAGE=firmware (1).image
 -4 herbie: some other quoting behaviour
```

The problem in the repo is currently always the whole string is quoted - works NOT for 2+4
Earlier before "quoting" commits, NOT working for 3

**Comment by fda77** [comment]
* ac2186a1c Makefile: revert e3cce9f9

Da das auf jeden Fall falsch ist hab ich es rückgägig gemacht. für die labor(1) muss dann noch was anderes her

**Reply by fda77** [reply]
* 545c7ce93 another quoting try

**Comment by Herbie-2005** [comment, marked as answer]
It works for me at the moment.

**Reply by fda77** [reply]
Thx for feedback, i tested 3 of the 4 cases and if wir works on your system too all is done
